resolwe_bio/processes/import_data/seq_reads.yml

```diff
@@ -218,3 +218,248 @@
+      docker:
+        image: resolwebio/legacy:latest
+    resources:
+      network: true
+      secrets: true
+  data_name: 'BaseSpace {{ file_ids.0|default("?") }}'
+  version: 0.0.1
+  type: data:reads:fastq:single
+  flow_collection: sample
+  category: upload
+  persistence: RAW
+  description: >
+    Import NGS reads in FASTQ format from BaseSpace.
+  input:
+    - name: file_ids
+      label: BaseSpace NGS reads (FASTQ) file IDs
+      type: list:basic:string
+    - name: access_token_secret
+      label: BaseSpace access token
+      type: basic:secret
+      description: >
+        BaseSpace access token secret handle needed to download the files.
+  output:
+    - name: fastq
+      label: Reads file
+      type: list:basic:file
+    - name: fastqc_url
+      label: Quality control with FastQC
+      type: list:basic:file:html
+    - name: fastqc_archive
+      label: Download FastQC archive
+      type: list:basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      {% for file_id in file_ids %}
+        BASESPACE_FILE_ARGUMENT+='--file-id {{file_id}} '
+      {% endfor %}
+
+      SECRET_PATH=/secrets/{{ access_token_secret.handle }}
+      basespace_download.py ${BASESPACE_FILE_ARGUMENT} --access-token-secret-path ${SECRET_PATH}
+      re-checkrc "Failed downloading files from BaseSpace."
+
+      gzip -dk *.gz
+
+      FASTQ=(*.fastq)
+      echo "Postprocessing FastQC..."
+      mkdir "fastqc" && fastqc "${FASTQ[@]}" --extract --outdir="fastqc" 2> stderr.txt
+      re-checkrc "Failed while processing with FastQC."
+      if [[ $(grep --text "Failed to process file" stderr.txt) != "" ]]
+      then
+        re-error "Failed while processing with FastQC."
+      fi
+
+      mv fastqc/*_fastqc.zip .
+
+      for report_dir in ./fastqc/*/
+      do
+        READS_NAME=`basename "${report_dir}" _fastqc`
+        ENCODING=`parse_encoding_type.py "${report_dir}fastqc_data.txt"`
+
+        if [ "${ENCODING}" == "Illumina 1.5" ] ||  [ "${ENCODING}" == "Illumina 1.3" ]; then
+
+          echo "Recoding input reads from Phred64 encoding to Phred33 encoding."
+          mv "${READS_NAME}.fastq" input_reads.fastq
+          TrimmomaticSE input_reads.fastq reformated.fastq.gz TOPHRED33
+          mv reformated.fastq.gz "${READS_NAME}.fastq.gz"
+
+        elif [ "${ENCODING}" != "Sanger / Illumina 1.9" ]; then
+          re-error "Only Sanger / Illumina 1.9 / llumina 1.5 / Illumina 1.3 encoding is supported."
+        fi
+      done
+
+      FASTQGZ=(*.fastq.gz)
+      re-save-file-list fastq "${FASTQGZ[@]}"
+
+      FASTQC=(*_fastqc.zip)
+      FASTQC_URL=()
+
+      for fqc in "${FASTQC[@]}"
+      do
+        NAME=`basename "$fqc" _fastqc.zip`
+        FASTQC_URL+=("{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}")
+      done
+
+      re-save-file-list fastqc_archive "${FASTQC[@]}"
+      re-save-list fastqc_url "${FASTQC_URL[@]}"
+
+- slug: basespace-fastq-paired
```

Comment:
Big security-related warning sign as above.

Additionally, this is duplicating basespace import functionality, just because the post-processing step is different. This is another case for why splitting this into two processes and encapsulating the two steps into a workflow should be considered for the future (when we implement workflow features that will make this easier).